opengovsg/GoGovSG
# Fix Pre-push hook to disallow improper formatting

## Problem
The current pre-push hook runs npm run lint which runs eslint with the --fix flag. This allows the linter to run successfully even though the pushed code contains formatting issues that are fixed by the --fix flag. Since the commit has already been done, the flag does nothing to truly resolve the issue.

Closes [#75]

## Solution

- Changed the `lint`/`lint-ts`/`lint-js` scripts to exclude the `--fix` flag.
- Added `lint:fix`/`lint-ts:fix`/`lint-js:fix` which include the `--fix` flag.

This way, the pre-push hook will be running the linter without the `--fix` flag and disallow improper formatting from getting pushed.

## Tests

1. Change any .ts/.js file to have 4 space indentation
2. Create a commit
3. Attempt to push to upstream (Github)
4. The pre-push commit should exit with a non-zero exit code and the code should not be pushed.
